### PR TITLE
Allow choosing start date to be when previous insurance expires

### PR DESCRIPTION
**Pull request by hedvigoscar**
There's one outstanding issue in this PR: How will the backend respond if we send null on editQuote for startDate? Will it ignore it, or will it explicitly set the data to be null?

EDIT: I've updated this code to use `removeStartDate` 🙂

### CODE REVIEW

**Comment by palmenhq** [comment]
I think rather you should use the `removeStartDate` mutation @hedvigoscar

File: src/pages/OfferNew/Introduction/Sidebar/StartDate.tsx [hunk 1]
```diff
@@ -236,10 +236,11 @@ export const StartDate: React.FC<Props> = ({
           setStartDate({
             variables: {
               quoteId: offerId,
-              date: format(newDateValue, gqlDateFormat),
+              date: newDateValue ? format(newDateValue, gqlDateFormat) : null,
```
Comment (palmenhq) [inline src/pages/OfferNew/Introduction/Sidebar/StartDate.tsx]: this'll most likely not work, we should use `removeStartDate` instead

**Comment by hedvigoscar** [comment]
Fantastic! That answers a big question I had, thank you 😄

File: src/new-components/DateInput/index.tsx [hunk 2]
```diff
@@ -345,6 +355,18 @@ export const DateInput: React.FC<DateInputProps> = ({
           }}
           render={(dayzedData) => <Calendar {...dayzedData} />}
         />
+        {hasCurrentInsurer ? (
```
Comment (palmenhq) [inline src/new-components/DateInput/index.tsx]: `hasCurrentInsuerer && (...)` *

**Comment by hedvigoscar** [comment]
@palmenhq Ready for re-review 🙂

File: src/new-components/DateInput/index.tsx [hunk 3]
```diff
@@ -344,6 +354,18 @@ export const DateInput: React.FC<DateInputProps> = ({
...
         />
+        {hasCurrentInsurer && (
+          <AtStartDateContainer>
+            <TextButton
+              onClick={() => {
+                setDate(null)
+                setOpen(false)
+              }}
+            >
+              När min gamla försäkring löper ut
```
Comment (gustafguner) [inline src/new-components/DateInput/index.tsx]: Text key police 🚓

Comment (hedvigoscar) [reply src/new-components/DateInput/index.tsx]: 🕵

**Pull request merged**